libs/npm/core/src/components/grid-table/Pager.tsx

```diff
@@ -8,95 +8,112 @@ import { Text } from '../form/text';
+  fromLocalStorage?: boolean;
+  /** The key name of the pageSize parameter. */
+  key?: string;
+}
+
+/** Pager properties. */
 export interface IPagerProps {
-  /**
-   * Page index position.
-   */
+  /** Page index position. */
   pageIndex: number;
-  /**
-   * Number of rows per page.
-   */
+  /** Number of rows per page. */
   pageSize: number;
-  /**
-   * Total number of pages.
-   */
+  /** Total number of pages. */
   pageCount: number;
-  /**
-   * Manual control for page sizing
-   */
-  manualPageSize?: boolean;
-  /**
-   * Number of pages to show in pager.
-   */
+  /** Number of pages to show in pager. */
   pageLimit?: number;
-  /**
-   * Show paging group buttons.
-   */
+  /** Show paging group buttons. */
   showPageGroup?: boolean;
-  /**
-   * Can go to previous page.
-   */
+  /** Can go to previous page. */
   canPreviousPage: boolean;
-  /**
-   * Can go to next page.
-   */
+  /** Can go to next page. */
   canNextPage: boolean;
-  /**
-   * An array of zero-based index integers corresponding to available pages in the table.
-   */
+  /** An array of zero-based index integers corresponding to available pages in the table. */
   pageOptions: number[];
+  /** Page size options */
+  pageSizeOptions?: IPageSizeOptions;
+  /** Event when changing page number. */
   gotoPage: (updater: number | ((pageIndex: number) => number)) => void;
+  /** Event when going to next page. */
   nextPage: () => void;
+  /** Event when going to previous page. */
   previousPage: () => void;
+  /** Event when page size changes. */
   setPageSize: (pageSize: number) => void;
 }
 
+/**
+ * Provides a manually change the page.
+ * @param param0 Component properties.
+ * @returns Component.
+ */
 export const Pager: React.FC<IPagerProps> = ({
   pageIndex,
   pageSize,
   pageCount,
-  pageLimit = 5,
+  pageLimit = 10,
   canPreviousPage,
   showPageGroup = false,
   canNextPage,
   pageOptions = [],
+  pageSizeOptions = {},
   gotoPage,
   nextPage,
   previousPage,
   setPageSize,
-  manualPageSize,
 }) => {
   const { search } = useLocation();
   let query = React.useMemo(() => new URLSearchParams(search), [search]);
   let [searchParams, setSearchParams] = useSearchParams(query);
 
-  React.useEffect(() => {
-    if (manualPageSize) {
-      const qty = searchParams.get(QTY);
-      const cachedQty = localStorage.getItem(QTY);
+  const [quantity, setQuantity] = React.useState(pageSize);
 
-      if (!!qty && qty !== pageSize.toString()) {
-        setPageSize(parseInt(qty));
-      }
+  const {
+    show: showPageSize = true,
+    fromUrl = true,
+    fromLocalStorage = false,
+    key = QTY,
+  } = pageSizeOptions;
 
-      if (!qty && !!cachedQty && cachedQty !== pageSize.toString()) {
-        setPageSize(parseInt(cachedQty));
-        setSearchParams({ ...searchParams, qty: cachedQty });
+  const qtyUrl = Number(searchParams.get(key));
+  const qtyLS = Number(localStorage.getItem(key));
+
+  React.useEffect(() => {
```

Comment:
This makes sure both local storage and the URL are up-to-date with the page size.